site/app/entities/course/CourseMaterial.php

```diff
@@ -72,7 +72,19 @@ class CourseMaterial {
     #[ORM\Column(type: Types::STRING, nullable: true)]
     protected ?string $title;
 
-    public function __construct(int $type, string $path, \DateTime $release_date, bool $hidden_from_students, float $priority, ?string $url, ?string $title) {
+    #[ORM\Column(type: Types::STRING, nullable: true)]
+    protected ?string $uploaded_by;
+
+    #[ORM\Column(type: Types::DATETIMETZ_MUTABLE, nullable: true)]
+    protected ?DateTime $upload_date;
+
+    #[ORM\Column(type: Types::STRING, nullable: true)]
+    protected ?string $last_edit_by;
+
+    #[ORM\Column(type: Types::DATETIMETZ_MUTABLE, nullable: true)]
+    protected ?DateTime $last_edit_date;
+
+    public function __construct(int $type, string $path, \DateTime $release_date, bool $hidden_from_students, float $priority, ?string $url, ?string $title, ?string $uploaded_by, ?\DateTime $upload_date, ?string $last_edit_by, ?\DateTime $last_edit_date) {
```

Comment:
Since you're editing this function's header, it would be good  to add a function level comment, as we're slowly trying to comment old functions.

Reply:
I will update the function header as soon as possible. Currently, I'm occupied with some other work.

Reply:
Added the comment for the function header @NiaHeermance.